packages/frontend/src/pages/common/pagination/ServerPagination.tsx

```diff
@@ -24,7 +26,9 @@ export function ServerPagination({
       page: page.toString(),
       perPage: perPage.toString(),
     })
-    return `${baseUrl}?${params}`
+    return baseUrl.includes('?')
+      ? `${baseUrl}&${params}`
+      : `${baseUrl}?${params}`
```

Comment:
To include previous params you should use URLSearchParams to parse the baseUrl and then construct new params that include the old ones.